lib/repo/repository.rb

```diff
@@ -149,73 +162,104 @@ def get_revision_by_timestamp(timestamp, path = nil)
       raise NotImplementedError,  "Repository.get_revision_by_timestamp: Not yet implemented"
     end
 
-    # Adds a user with a given permission-set to the repository
-    def add_user(user_id, permissions)
-      raise NotImplementedError,  "Repository.add_user: Not yet implemented"
+    def self.get_full_access_users
+      Admin.pluck(:user_name)
     end
 
-    # Removes user permissions for read & write access to the repository
-    def remove_user(user_id)
-      raise NotImplementedError,  "Repository.remove_user: Not yet implemented"
-    end
+    # Gets a list of users with permission to access the repo.
+    # Returns nil if there aren't any.
+    # All permissions are rw for the time being
+    def get_users
 
-    # Gets a list of users with permissions in question on the repository
-    #   use "Repository::Permission::ANY" to get a list of all users with any permissions
-    #   i.e. all users with at least read permissions
-    def get_users(permissions)
-      raise NotImplementedError, "Repository.get_users: Not yet implemented"
-    end
+      unless MarkusConfigurator.markus_config_repository_admin? # are we admin?
+        raise NotAuthorityError.new('Unable to get permissions: Not in authoritative mode!')
+      end
+      repo_name = get_repo_name
+      permissions = self.get_all_permissions
+      users = permissions[repo_name] + self.get_full_access_users
 
-    # Gets permissions for a particular user
-    def get_permissions(user_id)
-      raise NotImplementedError, "Repository.get_permissions: Not yet implemented"
+      return if users.empty?
```

Comment:
This isn't type-safe (if this returns `nil`, not compatible with usage below). Better to just return an empty list, i.e., just `users`.